indexer/db/db.go

```diff
@@ -401,68 +399,79 @@ func (d *Database) GetDepositsByAddress(address common.Address, page PaginationP
+		l1_blocks.number, l1_blocks.timestamp
+	FROM state_batches
+	INNER JOIN l1_blocks ON state_batches.block_hash = l1_blocks.hash
+	WHERE size + prev_total >= (
+		SELECT
+			number
+		FROM
+		withdrawals
+		INNER JOIN l2_blocks ON withdrawals.block_hash = l2_blocks.hash where tx_hash=$1
+	) ORDER BY "index" LIMIT 1;
 	`
 
-	var withdrawal *WithdrawalJSON
+	var batch *StateBatchJSON
 	err := txn(d.db, func(tx *sql.Tx) error {
-		row := tx.QueryRow(selectWithdrawalStatement, hash.String())
+		row := tx.QueryRow(selectWithdrawalBatchStatement, hash.String())
 		if row.Err() != nil {
 			return row.Err()
 		}
 
-		var l2Token Token
-		if err := row.Scan(
-			&withdrawal.GUID, &withdrawal.FromAddress, &withdrawal.ToAddress,
-			&withdrawal.Amount, &withdrawal.TxHash, &withdrawal.Data,
-			&withdrawal.L1Token, &l2Token.Address,
-			&l2Token.Name, &l2Token.Symbol, &l2Token.Decimals,
-			&withdrawal.L1BlockNumber, &withdrawal.L1BlockTimestamp,
-			&withdrawal.L2BlockNumber, &withdrawal.L2BlockTimestamp,
-		); err != nil {
+		var index, size, prevTotal, blockNumber, blockTimestamp uint64
+		var root, blockHash string
+		var extraData []byte
+		err := row.Scan(&index, &root, &size, &prevTotal, &extraData, &blockHash,
+			&blockNumber, &blockTimestamp)
+		if err != nil {
+			if errors.Is(err, sql.ErrNoRows) {
+				batch = nil
+				return nil
+			}
 			return err
 		}
-		withdrawal.L2Token = &l2Token
+
+		batch = &StateBatchJSON{
+			Index:          index,
+			Root:           root,
+			Size:           size,
+			PrevTotal:      prevTotal,
+			ExtraData:      extraData,
+			BlockHash:      blockHash,
+			BlockNumber:    blockNumber,
+			BlockTimestamp: blockTimestamp,
+		}
 
 		return nil
 	})
 	if err != nil {
 		return nil, err
 	}
 
-	return withdrawal, nil
+	return batch, nil
 }
 
 // GetWithdrawalsByAddress returns the list of Withdrawals indexed for the given
 // address paginated by the given params.
 func (d *Database) GetWithdrawalsByAddress(address common.Address, page PaginationParam) (*PaginatedWithdrawals, error) {
-	const selectWithdrawalsStatement = `
+	selectWithdrawalsStatement := fmt.Sprintf(`
 	SELECT
 	    withdrawals.guid, withdrawals.from_address, withdrawals.to_address,
 		withdrawals.amount, withdrawals.tx_hash, withdrawals.data,
 		withdrawals.l1_token, withdrawals.l2_token,
 		l2_tokens.name, l2_tokens.symbol, l2_tokens.decimals,
-		l2_blocks.number, l2_blocks.timestamp
+		l2_blocks.number, l2_blocks.timestamp, withdrawals.br_withdrawal_hash
 	FROM withdrawals
-		INNER JOIN l2_blocks ON withdrawals.l2_block_hash=l2_blocks.hash
+		INNER JOIN l2_blocks ON withdrawals.block_hash=l2_blocks.hash
 		INNER JOIN l2_tokens ON withdrawals.l2_token=l2_tokens.address
-	WHERE withdrawals.from_address = $1 ORDER BY l2_blocks.timestamp LIMIT $2 OFFSET $3;
-	`
+	WHERE withdrawals.from_address = $1 %s ORDER BY l2_blocks.timestamp LIMIT $2 OFFSET $3;
+	`, FinalizationStateAny.SQL())
```

Comment:
This is safe from sql injection attacks?

Reply:
Yes. FinalizationState is an int that returns a SQL fragment. In later PRs it’ll be used by the API to request withdrawals in different states. For not it just supports the one.

Reply:
It doesn’t consume any user input.

Reply:
Ok, i see that there is a switch statement that determines the possible strings that can be returned here